test/Microsoft.ML.Tests/ImagesTests.cs

```diff
@@ -184,6 +184,78 @@ public void TestGreyscaleTransformImages()
             Done();
         }
 
+        [Fact]
+        public void TestGrayScaleInMemory()
+        {
+            var imagesDataFile = SamplesUtils.DatasetUtils.DownloadImages();
+
+            var data = ML.Data.CreateTextLoader(new TextLoader.Options()
+            {
+                Columns = new[]
+                {
+                    new TextLoader.Column("ImagePath", DataKind.String, 0),
+                    new TextLoader.Column("Name", DataKind.String, 1),
+             }
+            }).Load(imagesDataFile);
+
+            var imagesFolder = Path.GetDirectoryName(imagesDataFile);
+            // Image loading and conversion pipeline.
+            var pipeline = ML.Transforms.LoadImages("ImageObject", imagesFolder, "ImagePath")
+                           .Append(ML.Transforms.ConvertToGrayscale("Grayscale", "ImageObject"));
+
+            // Test path: image files -> IDataView -> Enumerable of Bitmaps.
+            var transformedData = pipeline.Fit(data).Transform(data);
+            var transformedDataPoints = ML.Data.CreateEnumerable<TransformedImageDataPoint>(transformedData, true).ToList();
```

Comment:
>true [](start = 109, length = 4)

you're realizing the enumerable as a list, so reuse option should be false, otherwise your list will end up containing the same image (the last one).